src/backend/PortabilityService.Functions/Report.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.Azure.WebJobs.Host;
+using System;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+
+namespace PortabilityService.Functions
+{
+    public static class Report
+    {
+        [FunctionName("report")]
+        public static HttpResponseMessage Run(
+            [HttpTrigger(AuthorizationLevel.Function, "get", Route = "report/{submissionId}")] HttpRequestMessage req,
+            string submissionId,
+            TraceWriter log)
+        {
+            if (!ValidAccessKey(req))
+            {
+                return req.CreateResponse(HttpStatusCode.Unauthorized);
+            }
+
+            // simulate report generation taking some time
+            if (new Random().Next(10) < 4)
+            {
+                return req.CreateResponse(HttpStatusCode.NotFound);
+            }
+
+            switch (req.Headers.Accept.ToString())
+            {
+                case "application/json":
```

Comment:
Hmm... Do you know if there is a constants class with these mime types? I see it being used here and in ReportFormat.cs and wonder if they can be consolidated.

Reply:
To my knowledge, no. I added one we can use for now.